src/attack.rs

```diff
@@ -343,15 +373,53 @@ fn enemy_attack(
     }
 }
 
-fn attack_cleanup(query: Query<(Entity, &AttackTimer), With<Attack>>, mut commands: Commands) {
+fn activate_hitbox(
+    attack_query: Query<(Entity, &AttackFrames, &Parent), Without<Collider>>,
+    fighter_query: Query<&Animation, With<State>>,
+    mut commands: Commands,
+) {
+    for (entity, attack_frames, parent) in attack_query.iter() {
+        if let Ok(animation) = fighter_query.get(parent.0) {
+            if animation.current_frame >= attack_frames.startup
+                && animation.current_frame <= attack_frames.active
+            {
+                //TODO: insert Collider based on size and transform offset in attack asset
+                commands
+                    .entity(entity)
+                    .insert(Collider::cuboid(ATTACK_WIDTH * 0.8, ATTACK_HEIGHT * 0.8));
+            }
+        }
+    }
+}
+
+fn deactivate_hitbox(
+    query: Query<(Entity, &AttackFrames, &Parent), (With<Attack>, With<Collider>)>,
+    fighter_query: Query<&Animation, With<State>>,
+    mut commands: Commands,
+) {
+    for (entity, attack_frames, parent) in query.iter() {
+        if let Ok(animation) = fighter_query.get(parent.0) {
+            if animation.current_frame == attack_frames.recovery {
```

Comment:
In general design terms, should we account for lost frames (as a consequence, avoid exact matches where possible)? They're edge cases, so I don't know if it's worth or not.

Reply:
I agree that makes sense but also I would like to figure out a way to not have these sort of lost frames at all. In this case ill just replace this with `>=`.